Backends/CLX/medium.lisp

```diff
@@ -99,13 +109,18 @@
                      (sqrt (+ (expt x 2) (expt y 2))))))))
 
 (defun line-style-effective-thickness (line-style medium)
-  (round (* (line-style-thickness line-style)
-            (line-style-scale line-style medium))))
+  (* (line-style-thickness line-style)
+     (line-style-scale line-style medium)))
 
 (defun line-style-effective-dashes (line-style medium)
-  (let ((scale (line-style-scale line-style medium)))
-    (map 'list #'(lambda (dash) (round (* dash scale)))
-         (line-style-dashes line-style))))
+  (when-let ((dashes (line-style-dashes line-style)))
+    (let ((scale (line-style-scale line-style medium)))
+      (flet ((scale-dash (dash)
+               (* dash scale)))
+        (values (if (member dashes '(t nil))
```

Comment:
This could be `(eq dashes t)` since `(eq dashes nil)` is impossible in the `when-let` body.

Reply:
since there is no need to set dashes when it is solid, I've changed it once again to return one value:

```(defun line-style-effective-dashes (line-style medium)
  (when-let ((dashes (line-style-dashes line-style)))
    (let ((scale (line-style-scale line-style medium)))
      (if (eq dashes t)
          (* scale 3)
          (map 'list (lambda (d) (* scale d)) dashes)))))
```

and use it like this

```
(if-let ((dash-pattern (line-style-effective-dashes line-style medium)))
            (setf (xlib:gcontext-line-style gc) :dash
                  (xlib:gcontext-dashes gc) (if (atom dash-pattern)
                                                (round dash-pattern)
                                                (mapcar #'round dash-pattern)))
            (setf (xlib:gcontext-line-style gc) :solid))
```

Reply:
Right, returning two values was only necessary because we weren't sure whether `xlib:gcontext-dashes` has to be called in the `:solid` case as well.